### PR TITLE
Ability to send sms

### PR DESCRIPTION
**Pull request by elena-vi**

### CODE REVIEW

File: lib/hackney/income/send_email.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          recipient: recipient,
+          template_id: template_id,
+          reference: reference,
+          variables: variables
+        )
+      end
+
+      private
+
+      def reference_for(tenancy)
```
Comment (cob16) [inline lib/hackney/income/send_email.rb]: This method looks unused in email and SMS class

File: lib/hackney/income/send_sms.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        )
+        
+        # tenancy = Hackney::Income::Models::Tenancy.find_by(tenancy_ref: tenancy_ref)
+
+        # @events_gateway.create_event(
+        #   tenancy_ref: tenancy_ref,
+        #   type: 'sms_message_sent',
+        #   description: "Sent SMS message to #{contact_number_for(tenancy)}",
+        #   automated: false
+        # )
```
Comment (cob16) [inline lib/hackney/income/send_sms.rb]: Suggest deleting this commented code for now and add logging line instead

File: lib/hackney/income/use_case_factory.rb [hunk 3]
```diff
@@ -44,6 +62,15 @@ def assign_tenancy_to_user
 
       private
 
+      def notifications_gateway
+        Hackney::Income::GovNotifyGateway.new(
+          sms_sender_id: ENV['GOV_NOTIFY_SENDER_ID'],
+          # api_key: ENV['GOV_NOTIFY_API_KEY'],
```
Comment (cob16) [inline lib/hackney/income/use_case_factory.rb]: extra commented args can be removed

File: spec/controllers/messages_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    ).and_call_original
+
+    patch :get_templates, params: {type: 'sms'}
+
+    expect(response.body).to match(/sms/)
+  end
+
+end
+
+class StubGovNotifyGateway
```
Comment (cob16) [inline spec/controllers/messages_controller_spec.rb]: Suggest extracting this to a separate file

File: spec/lib/hackney/income/gov_notify_gateway_spec.rb [hunk 5]
```diff
@@ -0,0 +1,176 @@
+require 'rails_helper'
+
+describe Hackney::Income::GovNotifyGateway do
+  let(:sms_sender_id) { 'cool_sender_id' }
+  # let(:email_reply_to_id) { 'awesome_reply_to_email' }
```
Comment (cob16) [inline spec/lib/hackney/income/gov_notify_gateway_spec.rb]: delete commented code

File: spec/lib/hackney/income/gov_notify_gateway_spec.rb [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    end
+
+    after do
+      ENV.delete('SEND_LIVE_COMMUNICATIONS')
+    end
+  end
+
+  context 'when sending a text message to a tenant' do
+    before do
+      ENV['TEST_PHONE_NUMBER'] = '01234 123456'
```
Comment (cob16) [inline spec/lib/hackney/income/gov_notify_gateway_spec.rb]: This is a good example of why passing config into the gateway making testing easier.

**Comment by cob16** [comment]
Apologies in advance as much of my critique will before the code you didn't write as this is a port of code from the frontend.

File: config/routes.rb [hunk 7]
```diff
@@ -4,5 +4,8 @@
     get '/sync-cases', to: 'my_cases#sync'
     post '/users/find-or-create', to: 'users#create'
     patch '/tenancies/:tenancy_ref', to: 'tenancies#update'
+    post '/messages/send_sms', to: 'messages#send_sms'
+    post '/messages/send_email', to: 'messages#send_email'
+    get '/messages/get_templates', to: 'messages#get_templates'
```
[the remark this answers is not in the document]
Comment (elena-vi) [reply config/routes.rb]: this will need to be all refactored anyway so I'm suggesting leaving it for now

**Pull request merged**